send.c

```diff
@@ -351,7 +380,19 @@ static struct safe_buffer_list *add_ra_options_prefix(struct safe_buffer_list *s
 			} else {
 				if (cease_adv || schedule_option_prefix(dest, iface, prefix)) {
 					sbl = safe_buffer_list_append(sbl);
-					add_ra_option_prefix(sbl->sb, prefix, cease_adv);
+
+					/** We want to get the lowest value out of the configured lifetime (from /etc/radvd.conf) and the maximum lifetime on
+					 *  any address that is part of that prefix in the kernel to avoid advertising a prefix that might expire too soon */
+					unsigned int valid;
+					unsigned int preferred;
+					struct AdvPrefix xprefix = *prefix;
+
+					int ret = get_prefix_lifetimes (&xprefix, &valid, &preferred);
+					if (ret && valid < xprefix.curr_validlft && preferred < xprefix.curr_preferredlft) {
```

Comment:
This doesn't feel right, what if only one of them is less? Then the `if` doesn't fire.
What about introducing a helper that you can re-use it in all the spots:
```
static void limit_prefix_lifetimes(struct AdvPrefix const *prefix) {
  unsigned int valid, preferred;
  int ret = get_prefix_lifetimes (&prefix, &valid, &preferred);
  if(ret) {
    xprefix.curr_validlft = min(valid, xprefix.curr_validlft);
    xprefix.curr_preferredlft = min(preferred, xprefix.curr_preferredlft);
  }
}
```

I have a further worry reading your code, and really wish there was a good test framework: what about cases RADVD is adding the address on the local interface as well? Wouldn't the lifetimes decay to zero and never get increased again?